oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/LocalVocabularySelectField/LocalVocabularySelectField.jsx

```diff
@@ -113,16 +135,16 @@ export const LocalVocabularySelectField = ({
     );
   }
 
-  const serializedOptions = useMemo(
-    () =>
-      showLeafsOnly
-        ? serializedVocabularyItems(allOptions).filter(
-            (o) => o.element_type === "leaf"
-          )
-        : serializedVocabularyItems(allOptions),
+  let serializedOptions = useMemo(
+    () => processVocabularyItems(allOptions, showLeafsOnly, filterFunction),
     [allOptions, showLeafsOnly]
   );
 
+  let serializedFeaturedOptions = useMemo(
+    () => processVocabularyItems(allOptions, showLeafsOnly, filterFunction),
+    [featuredOptions, showLeafsOnly]
```

Comment:
filterFunction is also a dependency of useMeme

Reply:
@mirekys Hey, true, linter says this, but the problem is that you are going to be passing filter function usually inline as a prop i.e.    filterFunction={(options) =>
                options.filter((option) => option.props?.submission)
              }
              
 According to my understanding, this function would be different on each render, which in turn would make this memoization useless and expecting someone to use useCallback on this function passed as a prop seems unrealistic, so to be honest, I thought it was safe to leave the function out of the dependency array even though the linter complains. As I know the function is always that same function. I was not able to find a clever solution how to solve this. Do you maybe have some suggestions?

Reply:
>  ...this function would be different on each render...
If so, then the useMemo result would never update, when the filter function changes, which is bad :) I recall some solutions to this case should be in epicreact course?